chain-impl-mockchain/src/rewards.rs

```diff
@@ -276,13 +305,17 @@ mod tests {
                 }
                 NonZeroU32::new(number).unwrap()
             };
+            let npools_threshold = u32::arbitrary(g) + 1;
```

Comment:
this can overflow when `u32::arbitrary() == u32::max_value()`

Reply:
OK. I don't remember that code, but will check it.  There's no need for the threshold to be that large in practice.

Reply:
indeed that would work too

Reply:
I've put in a test that shouldn't overflow.  We'll need to merge the code back later, but will wait until I have everything stable first.